packages/svelte/src/compiler/migrate/index.js

```diff
@@ -729,9 +744,53 @@ const template = {
 		}
 		next();
 	},
+	SvelteSelf(node, { state, next }) {
+		const source = state.str.original.substring(node.start, node.end);
+		if (!state.filename) {
+			const indent = guess_indent(source);
+			state.str.prependRight(
+				node.start,
+				`<!-- @migration-task: migrate this by hand or call the migrate function with the filename of this file -->\n${indent}`
+			);
+			next();
+			return;
+		}
+		// by default we overwrite until the end of the node - 1
+		let end = node.end - 1;
+		// if it has some children we need to overwrite from start to start of the fragment
+		if (node.fragment.nodes.length > 0) {
```

Comment:
This seems a bit complicated, can you elaborate why? AFAIK we can just `str.update(node.start + 1, node.start + 1 + 'svelte:self'.length)`. For the end tag it needs a few more checks, that's true

Reply:
Duh... I'll add a post it on my screen with "check if there's a simpler way you idiot" 😁